compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -1782,10 +1787,14 @@ impl SharedEmitterMain {
             match message {
                 Ok(SharedEmitterMessage::Diagnostic(diag)) => {
                     let handler = sess.diagnostic();
-                    let mut d = rustc_errors::Diagnostic::new(diag.lvl, &diag.msg);
+                    let mut d = rustc_errors::Diagnostic::new(diag.lvl, String::new());
+                    d.message = diag.msg;
                     if let Some(code) = diag.code {
                         d.code(code);
                     }
+                    for (name, arg) in diag.args {
+                        d.set_arg(name, arg);
+                    }
```

Comment:
We could probably make a `d.replace_args` function that takes the whole map and just wholesale replaces its internal one with it - just to avoid some more cloning.

Reply:
Sounds great to me. I can make that function and use it here.